build.sbt

```diff
@@ -44,35 +42,61 @@ libraryDependencies ++= List(
   "org.scalatest" %% "scalatest" % "3.2.19" % Test
 )
 
-scalaVersion := "2.12.20"
+lazy val scala212 = "2.12.20" // bin/test-release.sh
+lazy val scala3 = "3.6.4" // bin/test-release.sh
+
+scalaVersion := scala3
+crossScalaVersions := Seq(scala212, scala3)
 
 // keep this as low as possible to avoid running into binary incompatibility such as https://github.com/sbt/sbt/issues/5049
-pluginCrossBuild / sbtVersion := "1.4.0"
+pluginCrossBuild / sbtVersion := {
+  scalaBinaryVersion.value match {
+    case "2.12" =>
+      "1.4.0"
+    case _ =>
+      "2.0.0-M4" // bin/test-release.sh
+  }
+}
 
 scriptedSbt := {
   val jdk = System.getProperty("java.specification.version").toDouble
 
   if (jdk >= 21)
-    "1.9.0" // first release that supports JDK21
+    Ordering[String].max(
+      (pluginCrossBuild / sbtVersion).value,
+      "1.9.0" // first release that supports JDK21
+    )
   else
     (pluginCrossBuild / sbtVersion).value
 }
 
-libraryDependencies += compilerPlugin(scalafixSemanticdb)
-
-scalacOptions ++= List("-Ywarn-unused", "-Yrangepos")
-
-scalacOptions ++= List(
-  "-target:jvm-1.8",
-  "-Xfatal-warnings",
-  "-Xlint"
-)
+scalacOptions ++= {
+  scalaBinaryVersion.value match {
+    case "2.12" =>
+      List(
+        "-Xsource:3",
+        "-Ywarn-unused",
+        "-Xlint",
+        "-Xfatal-warnings"
+      )
+    case _ =>
+      List(
+        "-Wunused:all",
+        "-Werror"
+      )
+  }
+}
 
 // Scripted
 enablePlugins(ScriptedPlugin)
 sbtPlugin := true
 scriptedBufferLog := false
-scriptedBatchExecution := true
+scriptedBatchExecution := {
+  scalaBinaryVersion.value match {
+    case "2.12" => true
+    case _ => false
```

Comment:
There is some contamination across scripted when batching in sbt 2.x
- `skip-windows/caching` -> `java.lang.ClassNotFoundException: LintAllPlugin$`
- `skip-sbt1.4/cross-build-scala3` -> `java.lang.ClassNotFoundException: LintAllPlugin$`
-`sbt-scalafix/scalafixOnCompile` -> `java.lang.ClassNotFoundException: FatalWarningsPlugin$`